tests/sharedv4/sharedv4_test.go

```diff
@@ -0,0 +1,269 @@
+		var frequency int
+		var timeout time.Duration
+		switch Inst().ChaosLevel {
+		case 10:
+			frequency = 100
+			timeout = 10 * time.Minute
+		case 9:
+			frequency = 90
+			timeout = 9 * time.Minute
+		case 8:
+			frequency = 80
+			timeout = 8 * time.Minute
+		case 7:
+			frequency = 70
+			timeout = 7 * time.Minute
+		case 6:
+			frequency = 60
+			timeout = 6 * time.Minute
+		case 5:
+			frequency = 50
+			timeout = 5 * time.Minute
+		case 4:
+			frequency = 40
+			timeout = 4 * time.Minute
+		case 3:
+			frequency = 30
+			timeout = 3 * time.Minute
+		case 2:
+			frequency = 20
+			timeout = 2 * time.Minute
+		case 1:
+			frequency = 10
+			timeout = 1 * time.Minute
+		default:
+			frequency = 10
+			timeout = 1 * time.Minute
+
+		}
+
+		customAppConfig := scheduler.AppConfig{
+			ClaimsCount: frequency,
+		}
+
+		provider := Inst().V.String()
+		contexts := []*scheduler.Context{}
+
+		Inst().CustomAppConfig["vdbench-sv4-svc-multivol"] = customAppConfig
+		err := Inst().S.RescanSpecs(Inst().SpecDir, provider)
+
+		Expect(err).NotTo(HaveOccurred(),
+			fmt.Sprintf("Failed to rescan specs from %s for storage provider %s. Error: [%v]",
+				Inst().SpecDir, provider, err))
+
+		Step("schedule application with multiple sharedv4-svc volumes attached", func() {
+			logrus.Infof("Number of Volumes to be mounted: %v", frequency)
+
+			taskName := "sharedv4-svc-multivol"
+
+			logrus.Infof("Task name %s\n", taskName)
+
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
+				newContexts := ScheduleApplications(taskName)
+				contexts = append(contexts, newContexts...)
+			}
+
+			for _, ctx := range contexts {
+				ctx.ReadinessTimeout = timeout
+				ctx.SkipVolumeValidation = false
+				ValidateContext(ctx)
+
+			}
+		})
+
+		Step("get nodes where app is running and restart volume driver", func() {
+			for _, ctx := range contexts {
+				appVolumes, err := Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				for _, appVolume := range appVolumes {
+					attachedNode, err := Inst().V.GetNodeForVolume(appVolume, defaultCommandTimeout, defaultCommandRetry)
+					Expect(err).NotTo(HaveOccurred())
+					Step(
+						fmt.Sprintf("stop volume driver %s on app %s's node: %s",
+							Inst().V.String(), ctx.App.Key, attachedNode.Name),
+						func() {
+							StopVolDriverAndWait([]node.Node{*attachedNode})
+						})
+
+					Step(
+						fmt.Sprintf("starting volume %s driver on app %s's node %s",
+							Inst().V.String(), ctx.App.Key, attachedNode.Name),
+						func() {
+							StartVolDriverAndWait([]node.Node{*attachedNode})
+						})
+
+					Step("Giving few seconds for volume driver to stabilize", func() {
+						time.Sleep(20 * time.Second)
+					})
+
+					Step(fmt.Sprintf("validate app %s", attachedNode.Name), func() {
+						ValidateContext(ctx)
+					})
+				}
+			}
+		})
+
+	})
+
+})
+
+// This test performs sharedv4 nfs server pod termination failover use case
+var _ = Describe("{NfsServerKill}", func() {
```

Comment:
Name is a bit misleading. I thought we are killing the NFS server process on the node. Instead, we are deleting the node itself. Maybe call it "NFSServerNodeDelete"? Also fix the comment.

Does this test work when running in non-aws environment? If not, do we need to skip it where it is not valid?

Reply:
Updated the name

It should work on non-aws setup also